static/js/message_list.js

```diff
@@ -413,6 +413,24 @@ exports.MessageList.prototype = {
         }
     },
 
+    update_locked_bookend: function MessageList_update_locked_bookend() {
```

Comment:
I think this function really belongs in `message_list_view.js`, since its main job is to draw.

Also, I guess the idea here is that we only put the "locked" bookend in narrows that are topic-specific, which seems sensible to me.  If that's so, we should probably put an explicit comment to that effect.  (If we wanted to support interleaved views, we'd need to think through that use case a bit.)

Lines 418 through 429 may still belong in `message_list.js`, and they could be something like `needs_topic_lock_bookend`.